src/metrics.jl

```diff
@@ -489,6 +489,8 @@ js_divergence(a::AbstractArray, b::AbstractArray) = JSDivergence()(a, b)
 
 result_type(dist::SpanNormDist, a::AbstractArray, b::AbstractArray) =
      typeof(eval_op(dist, oneunit(eltype(a)), oneunit(eltype(b))))
+result_type(dist::SpanNormDist, a::AbstractArray{<:AbstractArray}, b::AbstractArray{<:AbstractArray}) =
+     typeof(eval_op(dist, oneunit(eltype(first(a))), oneunit(eltype(first(b)))))
```

Comment:
```suggestion
     typeof(eval_op(dist, oneunit(eltype(eltype(a))), oneunit(eltype(eltype(b)))))
```

Reply:
Maybe we should rather override `result_type(dist::SpanNormDist, a::Type, b::Type)` and have global generic methods that call this? It sounds suboptimal to redefine methods for different inputs types.